app/src/main/java/org/mozilla/focus/menu/WebContextMenu.java

```diff
@@ -118,8 +120,22 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                         return true;
                     }
                     case R.id.menu_image_save: {
-                        final Download download = new Download(hitTarget.imageURL, null, null, null, -1, true);
-                        callback.onDownloadStart(download);
+                        if (URLUtil.guessFileName(hitTarget.imageURL, null, null).endsWith(".bin")) {
```

Comment:
Let's don't hardcode ".bin" here, Declare:
```
public static final DEFAULT_DOWNLOAD_EXTENSION = ".bin";
```
in `URLUtil` and change them respectively